src/core/server/http/router/router.ts

```diff
@@ -44,9 +44,12 @@ interface RouterRoute {
  *
  * @public
  */
-export type RouteRegistrar<Method extends RouteMethod> = <P, Q, B>(
+export type RouteRegistrar<
+  Method extends RouteMethod,
+  Context extends RequestHandlerContext = RequestHandlerContext
```

Comment:
Note that the order of generic params is different since `Context` is an optional parameter.